### PR TITLE
Baseline/atlasdb tests shared

### PR DESCRIPTION
**Pull request by gsheasby**
**Goals (and why)**: Add baseline for atlasdb-tests-shared

**Concerns (what feedback would you like?)**: Quite a few changes here - did a mistake creep in among the trivial changes?

**Priority (whenever / two weeks / yesterday)**: Today

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1873)

### CODE REVIEW

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java [hunk 1]
```diff
@@ -543,14 +549,18 @@ public void testGetAllTableNames() {
...
         assertEquals(0, keyValueService.getMetadataForTable(TEST_TABLE).length);
         keyValueService.putMetadataForTable(TEST_TABLE, AtlasDbConstants.GENERIC_TABLE_METADATA);
-        assertTrue(Arrays.equals(AtlasDbConstants.GENERIC_TABLE_METADATA, keyValueService.getMetadataForTable(TEST_TABLE)));
+        assertTrue(Arrays.equals(AtlasDbConstants.GENERIC_TABLE_METADATA,
+                keyValueService.getMetadataForTable(TEST_TABLE)));
     }
 
-    private static <V, T extends Iterator<RowResult<V>>> void assertRangeSizeAndOrdering(T it, int expectedSize, RangeRequest rangeRequest) {
+    private static <V, T extends Iterator<RowResult<V>>> void assertRangeSizeAndOrdering(T it,
+            int expectedSize, RangeRequest
```
Comment (tboam) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: Formatting here looks a bit funny.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java [hunk 2]
```diff
@@ -685,58 +712,78 @@ private void doTestGetRangePaging(int numColumnsInMetadata, int batchSizeHint, b
...
-        values.put(Cell.create("02".getBytes(), "c2".getBytes()), "e".getBytes());
-
-        values.put(Cell.create("03".getBytes(), "c1".getBytes()), "f".getBytes());
-
-        values.put(Cell.create("04".getBytes(), "c1".getBytes()), "g".getBytes());
-        values.put(Cell.create("04".getBytes(), RangeRequests.getLastRowName()), "h".getBytes());
-
-        values.put(Cell.create("05".getBytes(), "c1".getBytes()), "i".getBytes());
-        values.put(Cell.create(RangeRequests.getLastRowName(), "c1".getBytes()), "j".getBytes());
+        values.put(Cell.create("00".getBytes(StandardCharsets.UTF_8), "c1".getBytes(StandardCharsets.UTF_8)),
```
Comment (tboam) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: Understand you're trying to minimise the diff here, but I think this and the changes at the start of the file would look a lot better if we just had a helper method in this class for `utf8(String s)`.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java [hunk 3]
```diff
@@ -1192,8 +1249,10 @@ public void testGetRangeThrowsOnError() {
     @Test
     public void testGetRangeOfTimestampsThrowsOnError() {
         try {
-            keyValueService.getRangeOfTimestamps(TEST_NONEXISTING_TABLE, RangeRequest.all(), AtlasDbConstants.MAX_TS).hasNext();
-            Assert.fail("getRangeOfTimestamps must throw on failure");
+            assertFalse("getRangeOfTimestamps must throw on failure",
```
Comment (tboam) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: This change misses the case where `hasNext` returns false no longer failing the test.  I think the right thing is to not test the return value and to remove the `catch` and replace with an `expected=RuntimeException.class` on the `Test` annotation.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionTestSetup.java [hunk 4]
```diff
@@ -125,73 +127,78 @@ protected TransactionManager getManager() {
...
     }
 
-    protected void delete(Transaction t, String rowName, String columnName) {
-        t.delete(TEST_TABLE, ImmutableSet.of(Cell.create(PtBytes.toBytes(rowName), PtBytes.toBytes(columnName))));
+    protected void delete(Transaction txn, String rowName, String columnName) {
+        txn.delete(TEST_TABLE, ImmutableSet.of(Cell.create(PtBytes.toBytes(rowName), PtBytes.toBytes(columnName))));
     }
 
-    protected String get(Transaction t,
+    protected String get(Transaction txn,
```
Comment (tboam) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionTestSetup.java]: I think the diff here is correct, but would be easier to read if we kept the old ordering of methods.  Don't feel strongly that you should put it back though.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java (hunk 2 above)
Comment (gsheasby) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: Yep, it was tempting to add that kind of method. There's probably some kind of StringUtils class where it belongs (or maybe even already exists).

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionTestSetup.java (hunk 4 above)
Comment (gsheasby) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionTestSetup.java]: I was required to switch it around so that overloaded methods were together. Agreed that it could have been done in separate commits.
